datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -515,9 +520,14 @@ protected void processRecords(ConsumerRecords<?, ?> records, Instant readTime) {
     // send the batch out the other end
     translateAndSendBatch(records, readTime);
 
-    if (System.currentTimeMillis() - readTime.toEpochMilli() > _processingDelayLogThresholdMillis) {
+    long processingTimeMillis = System.currentTimeMillis() - readTime.toEpochMilli();
+    if (processingTimeMillis > _processingDelayLogThresholdMillis) {
       _consumerMetrics.updateProcessingAboveThreshold(1);
     }
+
+    if (_enableAdditionalMetrics) {
+      _consumerMetrics.updatePerEventProcessingTimeMs(records.count() == 0 ? 0 : processingTimeMillis / records.count());
```

Comment:
I dont think we will encounter a case where records count is 0 cause processRecords is called if records is not empty. Saying that its a good check to have.

Reply:
Yeah I did notice that, but just added the check to be extra careful. Will leave this as is for now. Thanks!